apps/src/aiComponentLibrary/userMessageEditor/UserMessageEditor.tsx

```diff
@@ -19,64 +19,75 @@ export interface UserMessageEditorProps {
-      e.preventDefault(); // Prevent the text box from having just a blank line.
-      handleSubmit(userMessage);
-    }
-  };
+    const userMessageIsEmpty = useMemo(() => {
+      return userMessage.trim() === '';
+    }, [userMessage]);
 
-  const handleSubmit = useCallback(
-    (userMessage: string) => {
-      onSubmit(userMessage);
-      setUserMessage('');
-    },
-    [onSubmit]
-  );
+    const handleKeyPress = (e: React.KeyboardEvent, userMessage: string) => {
+      if (e.key === 'Enter' && userMessage.trim() !== '') {
+        e.preventDefault(); // Prevent the text box from having just a blank line.
+        handleSubmit(userMessage);
+      }
+    };
 
-  const icon = {iconName: 'paper-plane'};
-  return (
-    <div
-      className={classnames(
-        moduleStyles.editorContainer,
-        editorContainerClassName
-      )}
-    >
-      <textarea
-        id="uitest-chat-textarea"
-        className={moduleStyles.textArea}
-        placeholder={customPlaceholder || commonI18n.aiUserMessagePlaceholder()}
-        onChange={e => setUserMessage(e.target.value)}
-        value={userMessage}
-        disabled={disabled}
-        onKeyDown={e => handleKeyPress(e, userMessage)}
-      />
+    const handleSubmit = useCallback(
+      (userMessage: string) => {
+        onSubmit(userMessage);
+        setUserMessage('');
+      },
+      [onSubmit]
+    );
 
-      <div className={moduleStyles.centerSingleItemContainer}>
-        <Button
-          id="uitest-chat-submit"
-          isIconOnly={!showSubmitLabel}
-          onClick={() => handleSubmit(userMessage)}
-          disabled={disabled || !userMessage || userMessageIsEmpty}
-          text={showSubmitLabel ? commonI18n.submit() : undefined}
-          {...{[showSubmitLabel ? 'iconLeft' : 'icon']: icon}}
+    const icon = {iconName: 'paper-plane'};
+    return (
+      <div
+        ref={ref}
+        className={classnames(
+          moduleStyles.editorContainer,
+          editorContainerClassName
+        )}
+      >
+        <textarea
+          id="uitest-chat-textarea"
+          className={moduleStyles.textArea}
+          placeholder={
+            customPlaceholder || commonI18n.aiUserMessagePlaceholder()
+          }
+          onChange={e => setUserMessage(e.target.value)}
+          value={userMessage}
+          disabled={disabled}
+          onKeyDown={e => handleKeyPress(e, userMessage)}
         />
+
+        <div className={moduleStyles.centerSingleItemContainer}>
+          <Button
+            id="uitest-chat-submit"
+            isIconOnly={!showSubmitLabel}
+            onClick={() => handleSubmit(userMessage)}
+            disabled={disabled || !userMessage || userMessageIsEmpty}
+            text={showSubmitLabel ? commonI18n.submit() : undefined}
+            {...{[showSubmitLabel ? 'iconLeft' : 'icon']: icon}}
+          />
+        </div>
       </div>
-    </div>
-  );
-};
+    );
+  }
+);
 
 export default UserMessageEditor;
```

Comment:
optional: to avoid reformatting the component, you could also just wrap the component with `forwardRef` when exporting, like

```
export default forwardRef<HTMLTextAreaElement, UserMessageEditorProps>(
  (props, ref) => <UserMessageEditor {...props} ref={ref} />
);
```
and just add `ref` to `UserMessageEditorProps` and attach to the `textarea` component

Reply:
As we discussed offline, this resulted in a warning when run locally.

<img width="677" alt="Screenshot 2024-09-20 at 4 29 28 PM" src="https://github.com/user-attachments/assets/4f5ed6d4-23ba-4eb4-be57-bd9cb89086b8">

Reply:
This is what I tried:
```
import classnames from 'classnames';
import React, {useState, useCallback, useMemo, forwardRef, Ref} from 'react';

import Button from '@cdo/apps/componentLibrary/button/Button';
import {commonI18n} from '@cdo/apps/types/locale';

import moduleStyles from './user-message-editor.module.scss';

/**
 * Renders the user message editor component.
 */

export interface UserMessageEditorProps {
  onSubmit: (userMessage: string) => void;
  disabled: boolean;
  showSubmitLabel?: boolean;
  /** Custom className for editor container */
  editorContainerClassName?: string;
  customPlaceholder?: string;
  ref?: Ref<HTMLTextAreaElement>;
}

const UserMessageEditor: React.FunctionComponent<UserMessageEditorProps> = ({
  onSubmit,
  disabled,
  editorContainerClassName,
  customPlaceholder,
  showSubmitLabel = false,
  ref,
}) => {
  const [userMessage, setUserMessage] = useState<string>('');

  const userMessageIsEmpty = useMemo(() => {
    return userMessage.trim() === '';
  }, [userMessage]);

  const handleKeyPress = (e: React.KeyboardEvent, userMessage: string) => {
    if (e.key === 'Enter' && userMessage.trim() !== '') {
      e.preventDefault(); // Prevent the text box from having just a blank line.
      handleSubmit(userMessage);
    }
  };

  const handleSubmit = useCallback(
    (userMessage: string) => {
      onSubmit(userMessage);
      setUserMessage('');
    },
    [onSubmit]
  );

  const icon = {iconName: 'paper-plane'};
  return (
    <div
      className={classnames(
        moduleStyles.editorContainer,
        editorContainerClassName
      )}
    >
      <textarea
        ref={ref}
        id="uitest-chat-textarea"
        className={moduleStyles.textArea}
        placeholder={customPlaceholder || commonI18n.aiUserMessagePlaceholder()}
        onChange={e => setUserMessage(e.target.value)}
        value={userMessage}
        disabled={disabled}
        onKeyDown={e => handleKeyPress(e, userMessage)}
      />

      <div className={moduleStyles.centerSingleItemContainer}>
        <Button
          id="uitest-chat-submit"
          isIconOnly={!showSubmitLabel}
          onClick={() => handleSubmit(userMessage)}
          disabled={disabled || !userMessage || userMessageIsEmpty}
          text={showSubmitLabel ? commonI18n.submit() : undefined}
          {...{[showSubmitLabel ? 'iconLeft' : 'icon']: icon}}
        />
      </div>
    </div>
  );
};

export default forwardRef<HTMLTextAreaElement, UserMessageEditorProps>(
  (props, ref) => <UserMessageEditor {...props} ref={ref} />
);

```